### PR TITLE
feat: implement consumer panel

### PR DESCRIPTION
**Pull request by matthew-chirgwin**
## Status
**READY**

Closes: #58

Do note however that #56 is still a WIP, so this will need revisiting

### Commit Message Title
* When opening this PR, the raiser should set the title of this PR to the first line of their desired commit message, e.g:
```
feat: implement consumer panel
```
* The reviewer should ensure that the first commit message field is of this form when performing the `squash and merge` from this page.

### Commit Message Description
```
- implements a consumer panel, with doc and tests
- brings in carbon theme ready for styling pass
- fixes few snags spotted during development
- simplifies Messages component, to make it more composable

Contributes to: #58

Signed-off-by: Matthew Chirgwin <chirmatt@uk.ibm.com>
```
* The reviewer should copy the above text into the extended description field when performing the `squash and merge` from this page.

## Checklist
- [X] Automated tests exist
- [X] Documentation exists [link]()
- [X] Local unit tests performed
- [X] Sufficient logging/trace
- [X] Desired commit message set as PR title and commit description set above

### CODE REVIEW

File: ui/src/Panels/Consumer/Consumer.view.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          disabled={!isReady}
+          onClick={onButtonClick}
+          {...idAttributeGenerator('consumer_button')}
+        >
+          {isRunning
+            ? translate('STOP_CONSUMING')
+            : translate('START_CONSUMING')}
+        </Button>
+      </div>
+      <div>
```
Comment (matthew-chirgwin) [inline ui/src/Panels/Consumer/Consumer.view.js:67]: This div will need replacing with the Consumer messages component when ready

File: ui/src/Panels/Consumer/Consumer.assets.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  controlledWebsocket,
+} from 'DevUtils/MockWebsocket';
+
+export { translations };
+
+export const consumerMockWebsocket = () =>
+  storybookWebsocket(CONSTANTS.CONSUMER);
+
+export const consumerMockWebsocketForTest = () =>
+  controlledWebsocket(CONSTANTS.CONSUMER);
```
Comment (ajborley) [inline ui/src/Panels/Consumer/Consumer.assets.js:14]: This file has assets for testing alongside assets needed in production - would it be worth splitting them out? In PR #100 I created a `Messages.spec.assets.js` file for use by both the `spec.js` and `stories.js` files. A production build could then tree-shake these files out.

File: ui/src/Panels/Consumer/Consumer.view.js (hunk 1 above)
Comment (ajborley) [reply ui/src/Panels/Consumer/Consumer.view.js:67]: Yep - looks like the ConsumerMessages component will need to be updated with the following:
 - add the message `index` property to the `messages` property PropTypes shapes.
- update the message `key` property to ``key={`consumed-message-${index}:${msg.index}`}``.
- add the `{...idAttributeGenerator('consumer_consumed_message')}` prop to the messages.

File: ui/src/Panels/Consumer/Consumer.assets.js (hunk 2 above)
Comment (matthew-chirgwin) [reply ui/src/Panels/Consumer/Consumer.assets.js:14]: We should be good here - we have set up the build to treeshake at the import/export level - so constant by constant, function by function, only what is used in prod will go out in the bundle

File: ui/src/Panels/Consumer/Consumer.view.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+          <ConsumerMessage
+            {...idAttributeGenerator('consumer_consumed_message')}
+            key={`consumed-message-${index}:${msg.index}`}
+            isFirst={index === 0}
+            error={
+              msg.status === CONSTANTS.VERTX_ERROR_STATUS
+                ? { message: translate('ERROR_CONSUMING', {}, true) }
+                : undefined
+            }
+            message={msg}
```
Comment (ajborley) [inline ui/src/Panels/Consumer/Consumer.view.js:78]: This also causes PropType warnings when the message is an error. Same remedy options as before applies here.

File: ui/src/Groups/Messages/Messages.view.js [hunk 4]
```diff
@@ -69,44 +26,32 @@ const renderEmptyMessagesPlaceholder = (translate, usage) => {
...
-      PropTypes.shape({
-        status: PropTypes.string.isRequired,
-        message: PropTypes.string,
-      }),
-    ])
-  ).isRequired,
-  /** optional - interaction handler function passed to the ConsumerMessage/ProducerMessage elements */
-  onInteraction: PropTypes.func,
+  /** optional - child content to render. If this is provided, the expectation is that it will contain either ProducedMessage or ConsumedMessage components. If none provided, empty state will render. */
+  children: PropTypes.node,
```
[the remark this answers is not in the document]
Comment (matthew-chirgwin) [reply ui/src/Groups/Messages/Messages.view.js:50]: Fair point - it could be either a single or array of elements (is what we really want). Will update!

**Pull request merged**